### ISSUE TITLE
slide 4

### ISSUE THREAD

**Issue by sembcorpp** [nus-cs2030/2122-s2#226]
# Description
Hi I think slide 4 of lecture 9 has an error it should print 1 instead of No Value for the ifPresentorElse example right?
Description of the question here.

## Topic:

Specific Topics this question is related to (If Any)

## Screenshots

Screenshots (if any)

**Comment by lukasleeyo** [comment]
![image](https://user-images.githubusercontent.com/19226196/161476520-dadce2fb-7c83-476a-be5c-794cdc49d43b.png)

Yea I also think the output for this code example is ``1`` and I typed it out in jshell just to confirm.

**Comment by yangyee-hub** [comment]
Yes that will be true and there is a error in th slide. Since after filtering, it will be Optional[1] that remains and thus 1 will be printed out.

**Comment by jiawei88** [comment]
Yeah I think that there is definitely an error in the slide as the value 1 should be printed. According to how ifPresentOrElse function works, since 1 will be filtered as the condition holds, there will be a value present and hence it is printed, and thus the second argument is not computed.

**Comment by tristky** [comment]
i was also utterly confused by the slide, thanks for clarifying
